---
title: "Effect of Sequencing Center and Population on Total DUF1220"
output: html_notebook
---
# Questions:  
- Should I be treating sequencing centers and populations as random rather than fixed? Are my models correct/do they make sense?

# Setup Code
```{r}
library(tidyverse)
library(car)
library(reshape2)
library(ggbeeswarm)

rd1000 <- read_delim("~/LabProjects/ReadDepth/data/1000Genomes/1000Genomes_with_unique_regions.txt", "\t")
colnames(rd1000) <- tolower(colnames(rd1000))

# Read in metadata file (sequence.index) to get info like population, sequencing center
metadata <- read_delim("~/LabProjects/ReadDepth/data/1000Genomes/sequence.index",delim = "\t", progress = FALSE)
colnames(metadata)[10] <- "sample"

# Merge the metatdata with the read depth data

# Filter data and select the columns we want to merge
metadata.tomerge <- metadata %>%
  filter(ANALYSIS_GROUP != "exome", INSTRUMENT_MODEL == "Illumina HiSeq 2000", LIBRARY_LAYOUT == "PAIRED") %>% filter(CENTER_NAME != "BCM") %>%
  group_by(sample, CENTER_NAME, POPULATION) %>%
  summarise(tot.reads = sum(READ_COUNT), mean.insert = mean(INSERT_SIZE))

colnames(metadata.tomerge) <- tolower(colnames(metadata.tomerge))

pop.groups <- read.delim("~/LabProjects/ReadDepth/data/1000Genomes/PopulationGroups.csv", sep = ",")
colnames(pop.groups) <- tolower(colnames(pop.groups))

# Merge metadata and population group information with read depth data

rd1000.data <- inner_join(rd1000, metadata.tomerge)
rd1000.data <- inner_join(rd1000.data, pop.groups)

## Summarise data by gene-specific clade groups
# Melt the data frame
rd1000.melt <- melt(rd1000.data, id.vars = c("sample", "center_name", "population", "pop.group", "mean.insert", "tot.reads"))
colnames(rd1000.melt)[7] <- "region"
colnames(rd1000.melt)[8] <- "rd"

# Write it to a file
write.table(rd1000.melt, file = "~/LabProjects/ReadDepth/variation_analysis/1000Genomes-rd-data-melted.txt", sep = "\t",
            row.names=FALSE,
            quote = FALSE)

```

```{bash}
# Executes the python code that adds information about gene-specific clade groups
python ~/LabProjects/ReadDepth/variation_analysis/for-making-gene-specific-clade-groups/scripts/make-gene-level-clade-groups3.py ~/LabProjects/ReadDepth/variation_analysis/1000Genomes-rd-data-melted.txt ~/LabProjects/ReadDepth/variation_analysis/for-making-gene-specific-clade-groups/supporting-files/clusters-collapse-hls.csv ~/LabProjects/G4/num-108bpHLS-per-gene.txt ~/LabProjects/ReadDepth/variation_analysis/
```

```{r}
# Read in the data with the gene specific clade group information added
rd1000.groups <- read_delim("~/LabProjects/ReadDepth/variation_analysis/1000Genomes-rd-data-with-groupings.txt", delim = "\t", col_names = FALSE)
colnames(rd1000.groups) <- c("sample","pop", "pop.group","center","mean.insert","tot.reads", "region", "gene.clade","value", "gene_copy", "group", "sample.group")

group.sums.by.sample <- rd1000.groups %>% filter(center != "BCM") %>%
  group_by(sample, pop, pop.group, center, mean.insert,tot.reads,group) %>% 
  summarise(group.total = sum(value))

total.duf.by.sample <- group.sums.by.sample %>% group_by(sample) %>%
  summarise(total.duf = sum(group.total))

group.sums.by.sample <- merge(group.sums.by.sample, total.duf.by.sample)

center.test <- group.sums.by.sample %>% mutate(BGI = ifelse(center == "BGI",1,0),
                                               SC = ifelse(center == "SC",1,0),
                                               BI = ifelse(center == "BI",1,0),
                                               ILLUMINA = ifelse(center == "ILLUMINA",1,0),
                                               MPIMG = ifelse(center == "MPIMG",1,0),
                                               WUGSC = ifelse(center == "WUGSC",1,0))

center.test <- center.test %>% mutate(ASW = ifelse(pop == "ASW",1,0),
                                      CEU = ifelse(pop == "CEU",1,0),
                                      CHB = ifelse(pop == "CHB",1,0),
                                      CHS = ifelse(pop == "CHS",1,0),
                                      CLM = ifelse(pop == "CLM",1,0),
                                      FIN = ifelse(pop == "FIN",1,0),
                                      GBR = ifelse(pop == "GBR",1,0),
                                      IBS = ifelse(pop == "IBS",1,0),
                                      JPT = ifelse(pop == "JPT",1,0),
                                      LWK = ifelse(pop == "LWK",1,0),
                                      MXL = ifelse(pop == "MXL",1,0),
                                      PUR = ifelse(pop == "PUR",1,0),
                                      TSI = ifelse(pop == "TSI",1,0),
                                      YRI = ifelse(pop == "YRI",1,0))

center.test <- center.test %>% select(-group, -group.total) 
center.test <- center.test %>% group_by(sample, pop, pop.group, center, mean.insert, tot.reads, BGI, SC, BI, ILLUMINA, MPIMG, WUGSC, ASW, CEU, CHB, CHS, CLM, FIN, GBR, IBS, JPT, LWK, MXL, PUR, TSI, YRI) %>% summarise(total.duf = median(total.duf)) 
```

Show the head of the data frame used for the below analysis
```{r}
head(center.test)
```

### Generate summary stats on number of samples from each population and sequencing center
#### Total number of samples (BCM excluded):  `r length(center.test$sample)`

#### Number of samples sequenced at each center
```{r}
num.samples.by.center <- center.test %>% group_by(center) %>% summarise(count = n()) %>% arrange(count)
ggplot(num.samples.by.center, aes(x = center, y = count, label = count)) + geom_bar(stat = "identity") + geom_text()
num.samples.by.center
```

##### Number of samples from each population
```{r}
num.samples.by.pop <- center.test %>% group_by(pop) %>% summarise(count = n()) %>% arrange(count)
ggplot(num.samples.by.pop, aes(x = pop, y = count, label = count)) + geom_bar(stat = "identity") + geom_text()
num.samples.by.pop
```

#### Number of samples from each population by sequencing center
```{r}
num.samples.by.pop.cent <- center.test %>% group_by(pop, center) %>% summarise(count = n()) %>% arrange(count)
ggplot(num.samples.by.pop.cent, aes(x = pop, y = count, fill = center, label = count)) + geom_bar(stat = "identity") + geom_text(size = 3, position = position_stack(vjust = 0.5))
num.samples.by.pop.cent
```

#### Number of samples from each sequencing center by population
```{r}
num.samples.by.cent.pop <- center.test %>% group_by(center,pop) %>% summarise(count = n()) %>% arrange(count)
ggplot(num.samples.by.cent.pop, aes(x = center, y = count, fill = pop, label = count)) + geom_bar(stat = "identity") + geom_text(size = 3, position = position_stack(vjust = 0.5))
num.samples.by.cent.pop %>% arrange(center, count, pop)
```


### Generate a model containing only sequencing centers - set center MPIMG as reference
_MPIMG set as reference because of most most linear association between ddPCR and RD for CON1, also, the mean total DUF1220 copy number for this center is intermediate to the others_
```{r}
reduced.cent <- lm(total.duf ~ BGI + SC + BI + ILLUMINA + WUGSC, center.test)
```
#### Look at normality of the outcome variable
```{r}
hist(center.test$total.duf)
```
Distribution of outcome looks normal <- Good

#### Look at normality of studentized residuals
```{r}
library(MASS)
hist(studres(reduced.cent))
detach("package:MASS", unload = T)
```
The residuals of each model appear normally distributed <- Good

#### Look for violations of homoscedascticity and linearity by sequencing center 
```{r}
crPlots(reduced.cent)
```
**Illumina group seems to violate homoscedasticity, probably due to small sample size.**

#### Remove ILLUMINA samples from datasets due to violation of homoscedasticity and regenerate model
```{r}
center.test <- center.test %>% filter(center != "ILLUMINA") 
reduced.cent <- lm(total.duf ~ BGI + SC + BI + WUGSC, center.test)
summary(reduced.cent)
```
**Conclusions:**  
Differences by sequencing center (not accounting for population) account for ~20% of the variation in Total DUF1220  

- Relative to MPIMG, no significant effect of being sequenced at SC or WUGSC
- Effect of being sequenced at BGI (relative to MPIMG) is, on average, an additional 72 copies of DUF1220  
- Effect of being sequenced at BI (relative to MPIMG) is, on average, an additional 39 copies of DUF1220  

#### Visualize squencing center associations
```{r}
ggplot(center.test, aes(x=center, y = total.duf)) + geom_point()
```



### Calculate model containing only populations - set population LWK as reference
_LWK selected as reference because the mean total DUF1220 copy number for this group is intermediate to the others_
```{r}
reduced.pop <- lm(total.duf ~ ASW + CEU + CHB + CHS + CLM + FIN + GBR + IBS + JPT + MXL + PUR + TSI + YRI, center.test)
```
#### Look at normality of studentized residuals
```{r}
library(MASS)
hist(studres(reduced.pop))
detach("package:MASS", unload = T)
```
#### Look for violations of homoscedascticity and linearity by population
_This is currently commented out because it keeps hanging up_
```{r}
#crPlots(reduced.pop)
```

### Examine results of model with populations only (LWK is reference)
```{r}
summary(reduced.pop)
```
**Conclusions:**  

In a model containing populations only (with LWK as reference):  
- Significant effect of being from CEU, JPT, PUR, or TSI (relative to LWK)  
- No other populations have significant effects relative to LWK  

#### Visualize population associations

```{r}
ggplot(center.test, aes(x = pop, y = total.duf)) + geom_point()
```


### Calculate model including all centers and populations (with sequencing center MPIMG and population LWK reference)
```{r}
full <- lm(total.duf ~ BGI + SC + BI + WUGSC + 
           ASW + CEU + CHB + CHS + CLM + FIN + GBR + IBS + JPT + MXL + PUR + TSI + YRI, center.test)
```

#### Look at normality of studentized residuals
```{r}
library(MASS)
hist(studres(full))
detach("package:MASS", unload = T)
```

### Examine results of full model (with sequencing center MPIMG and population LWK reference)
```{r}
summary(full)
```
#### Visualize relationship between population, sequencing center, and total duf1220

```{r}
ggplot(center.test,aes(x = pop, y = total.duf, color = center)) + geom_quasirandom()
```



**Conclusions:**  

In a model containing both sequencing center and population (with MPIMG and LWK as reference sequencing center and population respectively):    
- Effect of being sequenced at BGI and BI is still significant
- Only population that has a significant effect is CEU

## Test which model is best
### Compare a model with no predictors to a model with sequencing centers only
```{r }
no.pred <- lm(total.duf ~ 1, center.test)
anova(no.pred,reduced.cent)
```
**Conclusion:** Sequencing centers contribute significantly to the prediction of total DUF1220/Sequencing centers have a signficant effect on total DUF1220


### Compare a model with no predictors to a model with populations only
```{r}
anova(no.pred,reduced.pop)
```
**Conclusion:** Population contributes significantly to the prediction of total DUF1220/Population has a signficant effect on total DUF1220


### Compare a model with sequencing centers only to a model containing both sequencing centers and populations
```{r}
anova(reduced.cent,full)
```
**Conclusion:** The two models are significantly different and the full model (sequencing centers and populations) does a better job of predicting total DUF1220 than the model with sequencing center only

### Compare a model with populations only to a model containing both sequencing centers and populations
```{r}
anova(reduced.pop, full)
```
**Conclusion:**  
The two models are significantly different and the full model (sequencing centers and populations) does a better job of predicting total DUF1220 than the model with populations only

# Overall Conclusions:
- Large and significant effect of being sequenced at BGI or BI - on total DUF1220, remain after correcting for population.  
- For analysis of variation in population, consider excluding samples sequenced at those centers, or reporting the data for those centers separately from the data from the other three centers (MPIMG, SC, and WUGSC)
- Consider adding or removing samples from total sample set to equalize number of samples from each population and from each sequencing center

# Look into effects of coverage (total reads) and insert size
#### Effect of insert size
##### Some samples show "zero insert size", the data is "zero" in the sequence.index file -- removing those samples for this analysis since not sure what is going on there
```{r}
center.test <- center.test %>% filter(mean.insert != 0)
insert.model <- lm(total.duf ~ mean.insert, center.test)
summary(insert.model)
```
**Conclusion:**  
- Insert size has a significant effect of DUF1220  

##### Visualize association
```{r}
ggplot(center.test, aes(x = mean.insert, y = total.duf, color = center)) + geom_point()
```


#### Investigate a model that includes insert size and sequencing center
```{r}
center <- lm(total.duf ~ BGI + SC + BI + WUGSC, center.test) # need to regenerate the center model since I removed some datapoints (where mean insert size was 0)
insert.center <- lm(total.duf ~ BGI + SC + BI + WUGSC + mean.insert, center.test)
summary(insert.center)
```

##### Compare model of sequencing centers only to model with sequencing center and insert size
```{r}
anova(center, insert.center)
```
**Conclusion:**  
- A model with both sequencing centers and mean insert size is not better than a model with sequencing centers allone


```{r}
tot.reads.model <- lm(total.duf ~ tot.reads, center.test)
summary(tot.reads.model)
```
**Conclusion:**  
- Number of total reads is not linearly associated with Total DUF1220 

##### Visualize association
```{r}
ggplot(center.test, aes(x = tot.reads, y = total.duf, color = center)) + geom_point()
```

